pybamm/spatial_methods/finite_volume.py

```diff
@@ -135,22 +135,28 @@ def divergence(self, symbol, discretised_symbol, boundary_conditions):
         submesh = self.mesh[symbol.domain]
 
         divergence_matrix = self.divergence_matrix(symbol.domains)
+        coord_sys = submesh.coord_sys
 
         # check coordinate system
-        if submesh.coord_sys in ["cylindrical polar", "spherical polar"]:
+        if coord_sys in ["cylindrical polar", "spherical polar"]:
             second_dim_repeats = self._get_auxiliary_domain_repeats(symbol.domains)
             # create np.array of repeated submesh.edges
             r_edges_numpy = np.kron(np.ones(second_dim_repeats), submesh.edges)
             r_edges = pybamm.Vector(r_edges_numpy)
-            if submesh.coord_sys == "spherical polar":
+            if coord_sys == "spherical polar":
                 out = divergence_matrix @ ((r_edges**2) * discretised_symbol)
-            elif submesh.coord_sys == "cylindrical polar":
+            elif coord_sys == "cylindrical polar":
                 out = divergence_matrix @ (r_edges * discretised_symbol)
-        else:
+        elif coord_sys == "cartesian":
             out = divergence_matrix @ discretised_symbol
-
+        elif coord_sys is not None:
+            raise ValueError(
+                f"Coordinate system is {coord_sys}, not in"
+                " cartesian, cylindrical polar, spherical polar, None")
```

Comment:
Also note that this line is not covered so an additional test is needed.

Reply:
ok, that comes something I am very weak now. Just reading the test_finite_volume.py file and only come up with:

`    def test_divergence(self):
        whole_cell = ["negative electrode", "separator", "positive electrode"]
        var = pybamm.Variable("var", domain=whole_cell)
        x = pybamm.SpatialVariableEdge(
                "x_s",
                domain=["separator"],
                auxiliary_domains={"secondary": "current collector"},
                coord_sys="cartesian_2",
            )
        with self.assertRaises(ValueError):
            pybamm.SpatialMethod.divergence(var,x)`

Reply:
but I am pretty sure this won't work, so just to raise a question.

Reply:
This test is similar to the one you have, which is necessary because it checks the error is raised in `independent_variable.py`. You need another test that calls directly the `divergence` method in `finite_volume.py`. However, I am not sure we can do that test without triggering the other error before. That's because in order to define the symbol to pass to `divergence` we need to give it a domain with its own coordinate system. @kratman, any thoughts?

Reply:
Thank you for the kind reply. @brosaplanella That's something I am a bit struggled as well. Though I notice that I can get some mesh by  `mesh = get_mesh_for_testing()`. If I look closer at `get_mesh_for_testing`, they already assign coordinate system, and I don't know how to change that.